### PR TITLE
feat: list and replace event listeners

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-991

## Description :pencil:

Allow users to list and replace listeners for the different viewer events.

Also uses this in `MeasurementTool` to disable all other click handlers while measurement mode is active, and restoring them afterwards.

## Test instructions :information_source:

- Open examples, preferrably a remote model
- Click a node
- Notice console log with click info
- Start measurement mode
- Click node
- Notice that no click info is logged

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/utilities/src/events/EventTrigger.ts [hunk 1]
```diff
@@ -2,10 +2,25 @@
  * Copyright 2021 Cognite AS
  */
 
+/**
+ * A view into an event viewer trigger
+ */
+export interface EventTriggerView<TListener extends (...args: any[]) => void> {
```
Comment (Savokr) [inline viewer/packages/utilities/src/events/EventTrigger.ts:8]: I feel like FDM influenced name choice for that interface 😀. I think we can follow similar convention as we have with other interfaces in Reveal and rename it to old `EventTrigger` and then change the name of implementation class to something like `DefaultEventTrigger` or `InternalEventTrigger`.

File: viewer/packages/tools/src/Measurement/MeasurementTool.ts [hunk 2]
```diff
@@ -251,12 +260,24 @@ export class MeasurementTool extends Cognite3DViewerToolBase {
...
 
+  private restoreViewerClickHandlers(): void {
+    const triggerView = this._viewer.getEventHandlerTrigger('click');
+
+    // Make sure to include listeners added after the override so that they persist
+    const potentiallyAddedListeners = triggerView.getListeners().splice(1);
+
+    triggerView.replaceListeners([...this._cachedClickHandlers, ...potentiallyAddedListeners]);
+    this._cachedClickHandlers.splice(0);
+  }
```
Comment (Savokr) [inline viewer/packages/tools/src/Measurement/MeasurementTool.ts:279]: Not sure about the API here, maybe we can make it a little easier? If we just want to temporary disable all listeners why we expose their replacement? Can't we just make a `disableCurrentlyAdded` method or something similar. Then the code here will be much simpler, without caring about added listeners and other stuff.

File: viewer/packages/utilities/src/events/EventTrigger.ts [hunk 3]
```diff
@@ -2,10 +2,25 @@
...
+  /**
+   * Get all event listeners subscribed to this trigger currently
+   */
+  getListeners(): TListener[];
+
+  /**
+   * Replace all currently registered listeners
+   */
+  replaceListeners(listeners: TListener[]): TListener[];
```
Comment (Savokr) [inline viewer/packages/utilities/src/events/EventTrigger.ts:17]: API seems a little strange for me as mentioned in the comment for `MeasurementTool.ts` file

**Review by Savokr — COMMENTED** [review]
Works good but have a few doubts about newly exposed API for EventTrigger.

**Comment by haakonflatval-cognite** [comment]
We agreed that we will instead do this a somewhat more elegant way: https://cognitedata.atlassian.net/browse/UX-1291

**Pull request closed without merging**